web-animations-1/Overview.bs

```diff
@@ -6733,12 +6734,21 @@ The <code>Animatable</code> interface mixin {#the-animatable-interface-mixin}
 
   : <dfn lt="getAnimations(options)">sequence<Animation> getAnimations(|options|)</dfn>
   ::
-      Returns the set of [=relevant animations=] for this object,
-      or, if an {{Animatable/getAnimations(options)/options}} parameter
-      is passed with {{GetAnimationsOptions/subtree}} set to true,
-      returns the set of [=relevant animations for a subtree=] for this object.
+      1. Let |object| be the object on which this method was called.
 
-      The returned list is sorted using
+      1. If {{Animatable/getAnimations(options)/options}} is passed
+          with a non-null {{GetAnimationsOptions/pseudoElement}},
```

Comment:
We should probably have a similar SyntaxError exception thrown as we do for trying to create an animation effect targeting an invalid pseudo, e.g. see https://www.w3.org/TR/web-animations-1/#dom-keyframeeffect-pseudoelement

You can see this is also referenced by the KeyframeEffect constructor: https://www.w3.org/TR/web-animations-1/#ref-for-dom-keyframeeffect-pseudoelement, though in this case since we're not setting `pseudoElement` it probably can't point to the same reference.

Reply:
Done. I've tried to define the algorithm in the old spot and reference it, instead of copying the text. But let me know what you prefer. Also, do you think we need a separate resolution for the DOMException, since I don't believe getAnimations can throw prior to this change

Reply:
So looking at `getComputedStyle` which is very similar, it looks like MDN [says it can throw on an invalid selector](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle#exceptions) however I'm not sure where this is called out in the [spec](https://drafts.csswg.org/cssom/#dom-window-getcomputedstyle) or how to trigger an exception in chrome, whereas for element.animate it is easy to trigger the exception with `elem.animate({}, {pseudoElement: ':invalid-pseudo'});`.

So we should probably get a consistent story for when we should throw on invalid pseudo selectors, but maybe this should be a separate issue explaining the current behavior of the APIs we have using pseudoElement selectors.

Reply:
I'd propose for now we land this with the exception for self consistency within the same spec, and follow up on this.